packages/core/src/external/aws/lambda-logic/modify-resources.ts

```diff
@@ -0,0 +1,599 @@
+  AllergyIntolerance,
+  "status" | "clinicalStatus" | "reaction"
+> & {
+  status?: string | undefined;
+  context?: AllergyContext[] | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimAllergyIntolerance(res: AllergyIntolerance): SlimAllergyIntolerance {
+  const updRes = cloneDeep(res);
+  const status = Array.from(
+    new Set(updRes.clinicalStatus?.coding?.flatMap(coding => coding.code || []))
+  ).join(", ");
+
+  const context = updRes.reaction?.flatMap(reaction => {
+    const substance = getUniqueDisplaysString(reaction.substance);
+    const manifestations = getUniqueDisplaysString(reaction.manifestation);
+    if (!substance && !manifestations) return [];
+    return { substance, manifestations };
+  });
+
+  delete updRes.reaction;
+  delete updRes.clinicalStatus;
+
+  return {
+    ...updRes,
+    status: isUselessStatus(status) ? undefined : status,
+    context,
+  };
+}
+
+export type SlimImmunization = Omit<Immunization, "vaccineCode" | "site" | "route"> & {
+  vaccineCode?: string | undefined;
+  site?: string | undefined;
+  route?: string | undefined;
+  reference?: Record<string, Partial<SlimOrganization>>;
+};
+
+function slimImmunization(res: Immunization): SlimImmunization | undefined {
+  if (res.vaccineCode) {
+    const resVaccineCodeString = JSON.stringify(res.vaccineCode).toLowerCase();
+    if (
+      resVaccineCodeString.includes("no data") ||
+      resVaccineCodeString.includes("no immunization")
+    ) {
+      return undefined;
+    }
+  }
+
+  const updRes = cloneDeep(res);
+  // Remove the unwanted properties directly
+  delete updRes.lotNumber;
+  delete updRes.doseQuantity;
+
+  return {
+    ...updRes,
+    vaccineCode: getUniqueDisplaysString(res.vaccineCode),
+    site: res.site?.text,
+    route: getUniqueDisplaysString(res.route),
+  };
+}
+
+export type SlimPractitioner = Omit<Practitioner, "name" | "qualification" | "address"> & {
+  name?: string | undefined;
+  qualification?: string | undefined;
+  address?: string | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimPractitioner(res: Practitioner): SlimPractitioner {
+  const updRes = cloneDeep(res);
+  return {
+    ...updRes,
+    name: getNameString(res.name),
+    qualification: getLongestDisplay(res.qualification?.[0]?.code),
+    address: getAddressString(res.address),
+  };
+}
+
+export type SlimProcedure = Omit<Procedure, "name" | "status" | "bodySite"> & {
+  name?: string | undefined;
+  status?: string | undefined;
+  reference?: Record<string, string>;
+  bodySite?: string | undefined;
+};
+
+function slimProcedure(res: Procedure): SlimProcedure | undefined {
+  const name = getUniqueDisplaysString(res.code);
+  if (name?.includes("no data")) return undefined;
+
+  const bodySite = getUniqueDisplaysString(res.bodySite);
+  const updRes = cloneDeep(res);
+  delete updRes.code;
+  delete updRes.reasonCode; // TODO: #2510 - Introduce term server lookup here
```

Comment:
Did we try to leave these codes when there's no text representation and ask the LLM to fill them out? No change needed here, mostly to bring up the discussion.

Reply:
No, we haven't tried that. But it would be a good experiment to try, so I'll put it on my list / issue TODOs